cnd/src/btsieve/ethereum/mod.rs

```diff
@@ -192,71 +376,57 @@ fn seen_block_or_predates_start_of_swap(
     }
 }
 
-async fn check_block_against_pattern<C>(
-    connector: C,
-    block: Block,
-    pattern: TransactionPattern,
-) -> anyhow::Result<Option<TransactionAndReceipt>>
-where
-    C: ReceiptByHash<Receipt = Option<TransactionReceipt>, TransactionHash = Hash>,
-{
-    let needs_receipt = pattern.needs_receipts(&block);
-    let block_hash = block
-        .hash
-        .ok_or_else(|| anyhow::anyhow!("block without hash"))?;
+impl Predates for Block {
+    fn predates(&self, timestamp: NaiveDateTime) -> bool {
+        let unix_timestamp = timestamp.timestamp();
 
-    if needs_receipt {
-        tracing::debug!(
-            "bloom-filter of block {:x} suggests to fetch receipts for {:?}",
-            block_hash,
-            pattern
-        );
-    } else {
-        tracing::debug!(
-            "bloom-filter of block {:x} suggests to not fetch receipts for {:?}",
-            block_hash,
-            pattern
-        );
+        self.timestamp < U256::from(unix_timestamp)
     }
+}
 
-    for transaction in block.transactions.into_iter() {
-        let tx_hash = transaction.hash;
-
-        let receipt = connector
-            .receipt_by_hash(tx_hash)
-            .compat()
-            .await?
-            .ok_or_else(|| {
-                anyhow::anyhow!(
-                    "Could not get transaction receipt for transaction {:x}",
-                    tx_hash
-                )
-            })?;
-
-        let result = pattern.matches(&transaction, &receipt);
-
-        tracing::debug!(
-            "matching {:?} against transaction {:x} yielded {}",
-            pattern,
-            tx_hash,
-            result
-        );
-
-        if result {
-            return Ok(Some(TransactionAndReceipt {
-                transaction,
-                receipt,
-            }));
-        }
-    }
+#[derive(Clone, Copy, Default, Eq, PartialEq, serde::Serialize, serdebug::SerDebug)]
+#[serde(transparent)]
+pub struct Topic(pub H256);
 
-    Ok(None)
+/// Event works similar to web3 filters:
```

Comment:
Should update or delete these docs.

Also it would be nice to not `ignore` them if we keep them but just fix the compile errors in there.

Reply:
Had a play with this, did not remove the 'ignore'.  Seems to me that adding all the type stuff to satisfy the compiler just makes the comment more cluttered with no real added advantage (since we are only parsing static strings into Rust types).